packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,829 @@
+      }
+      logicalKeyId = shortcut.trigger.keyId;
+    }
+    return <String, Object?>{
+      _kIdKey: delegate.getId(item),
+      _kLabelKey: item.label,
+      _kEnabledKey: item.onSelected != null,
+      if (shortcut != null) _kShortcutTriggerKey: logicalKeyId,
+      if (shortcut != null) _kShortcutModifiersKey: modifiers,
+    };
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('label', label));
+    properties.add(DiagnosticsProperty<ShortcutActivator?>('shortcut', shortcut, defaultValue: null));
+    properties.add(FlagProperty('enabled', value: onSelected != null, ifFalse: 'DISABLED'));
+  }
+}
+
+/// A widget that represents a menu item that is preconfigured for the given
+/// platform.
+///
+/// This is used to add things like the "About" and "Quit" menu items to a
+/// platform menu.
+///
+/// The [type] enum determines which type of platform defined menu will be
+/// added.
+///
+/// This is most useful on a macOS platform where there are many different types
+/// of platform provided menu items in the standard menu setup.
+///
+/// In order to know if a [PlatformProvidedMenuItem] is available on a
+/// particular platform, call [PlatformProvidedMenuItem.hasMenu].
+///
+/// If the platform does not support the given [type], then the menu item will
+/// throw an [ArgumentError].
+///
+/// See also:
+///
+///  * [PlatformMenuBar] which takes these items for inclusion in a
+///    platform-rendered menu bar.
+class PlatformProvidedMenuItem extends PlatformMenuBarItem with DefaultPlatformMenuDelegateSerializer {
+  /// Creates a const [PlatformProvidedMenuItem] of the appropriate type. Throws if the
+  /// platform doesn't support the given default menu type.
+  ///
+  /// The [type] argument is required.
+  const PlatformProvidedMenuItem({
+    required this.type,
+    this.enabled = true,
+  }) : super(
+          label: '', // The label is ignored for standard menus.
+        );
+
+  /// The type of default menu this is.
+  ///
+  /// See [PlatformProvidedMenuItemType] for the different types available.  Not
+  /// all of the types will be available on every platform. Use [hasMenu] to
+  /// determine if the current platform has a given default menu item.
+  ///
+  /// If the platform does not support the given [type], then the menu item will
+  /// throw an [ArgumentError].
+  final PlatformProvidedMenuItemType type;
+
+  /// True if this [PlatformProvidedMenuItem] should be enabled or not.
+  final bool enabled;
+
+  /// Checks to see if the given default menu type is supported on this
+  /// platform.
+  static bool hasMenu(PlatformProvidedMenuItemType menu) {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.iOS:
+      case TargetPlatform.fuchsia:
+        return false;
+      case TargetPlatform.linux:
```

Comment:
I thought we only support PlatformMenus on MacOS? But there's some support for Linux?